futurex_openedx_extensions/helpers/tenants.py

```diff
@@ -369,3 +373,139 @@ def create_new_tenant_config(sub_domain: str, platform_name: str) -> TenantConfi
+    """
+
+    def get_nested_value(config: dict, path: str) -> Any:
+        """
+        Retrieves a value from a nested dictionary based on a comma-separated path.
+
+        :param config: The dictionary to search.
+        :param path: The comma-separated path to traverse.
+        :return: The found value or None if not found.
+        """
+        keys = [key.strip() for key in path.split(',')]
+        for key in keys:
+            if not isinstance(config, dict) or key not in config:
+                return None
+            config = config[key]
+        return config
+
+    if not config:
+        return None
+
+    if published_only:
+        return get_nested_value(config, path)
+
+    if draft_only:
+        return get_nested_value(config.get('config_draft', {}), path)
+
+    return get_nested_value(config.get('config_draft', {}), path) or get_nested_value(config, path)
+
+
+def get_draft_tenant_config(tenant_id: int) -> dict:
+    """
+    This function fetches the configuration for the specified tenant and returns the
+    fields that differ between the published and draft configurations.
+
+    :param tenant_id: The ID of the tenant whose draft configuration are to be retrieved.
+    :type tenant_id: int
+    :return: A dictionary containing updated fields with published and draft values.
+    :rtype: dict
+    """
+    config = get_all_tenants_info()['config'].get(tenant_id, {})
+    updated_fields = {}
+    for field_config in ConfigAccessControl.objects.all():
+        draft_value = get_tenant_config_value(config, field_config.path, published_only=False, draft_only=True)
+        published_value = get_tenant_config_value(config, field_config.path, published_only=True, draft_only=False)
+        if draft_value and published_value:
+            updated_fields[field_config.key_name] = {
+                'published_value': published_value,
+                'draft_value': draft_value,
+            }
+    return updated_fields
+
+
+def update_draft_tenant_config(
+    tenant_id: int, key_access_info: ConfigAccessControl, current_value: str, new_value: str, reset: bool = False
+) -> None:
+    """
+    Updates the draft configuration for the specified tenant with the new value.
+    If reset is True, it resets the field to its published value or removes it.
+
+    :param tenant_config: The configuration for the tenant, containing both published and draft data.
+    :type tenant_config: dict
+    :param key_access_info: A dictionary or path representing the nested key in the configuration.
+    :type key_access_info: dict or list of strings
+    :param new_value: The new value to update the field with.
+    :type new_value: str
+    :param reset: A boolean indicating whether to reset the field to its published value. Defaults to False.
+    :type reset: bool
+    :return: A dictionary with the updated fields and their new values.
+    :rtype: dict
+    """
+    config = get_all_tenants_info()['config'].get(tenant_id)
```

Comment:
well, this is one of the cases that we must not use the cache 🥲 . We need to keep safe editing in `lms_config`, so we'll have to always fetch from database